mailbox/interface.go

```diff
@@ -108,19 +108,26 @@ type connKit struct {
 
 	receiveSID [64]byte
 	sendSID    [64]byte
+
+	recvBuffer bytes.Buffer
 }
 
 // Read reads data from the underlying control connection.
 //
 // NOTE: This is part of the net.Conn interface.
 func (k *connKit) Read(b []byte) (int, error) {
-	data := NewMsgData(ProtocolVersion, nil)
-	if err := k.impl.ReceiveControlMsg(data); err != nil {
-		return 0, err
+	if k.recvBuffer.Len() == 0 {
```

Comment:
What's the implication of _not_ having this? The we violate the normal `Read` semantics and return potentially _more_ bytes that the caller expected (seen via the value of the `n` param).

Reply:
yeah so this ensures that we dont violate this rule.

The reason this was not required before is that all the acts were specific expected lengths and so we knew how big the byte slice we passed in here needed to be. But now, if we have a dynamic field in the Act 2 message, we will not know how big the slice needs to be and need to first read the length field. But since the entire payload is returned from gbn in one go, we need this cache buffer here.